smoke/IotEdgeQuickstart/details/Details.cs

```diff
@@ -334,6 +339,64 @@ protected async Task VerifyDataOnIoTHub(string moduleId)
             await eventHubClient.CloseAsync();
         }
 
+        protected async Task VerifyTwinAsync()
+        {
+            if (this.TwinTestFileName.HasValue)
+            {
+                string twinTestJson = JObject.Parse(File.ReadAllText(this.TwinTestFileName.GetOrElse(string.Empty))).ToString();
+
+                var twinTest = JsonConvert.DeserializeObject<TwinTestConfiguration>(twinTestJson);
+
+                Twin currentTwin = await this.context.RegistryManager.GetTwinAsync(this.context.Device.Id, twinTest.ModuleId);
+
+                if (twinTest.TwinTest.ContainsKey("desired") && twinTest.TwinTest["desired"].Count > 0)
+                {
+                    StringBuilder twinpatch = new StringBuilder(@"{ properties: { desired: { ");
+                    foreach (var desiredProperty in twinTest.TwinTest["desired"])
+                    {
+                        twinpatch.Append($@"{desiredProperty.Key} : {desiredProperty.Value},");
+                    }
+
+                    twinpatch.Append(@"} } }");
+
+                    // Build Patch Object.
+                    await this.context.RegistryManager.UpdateTwinAsync(this.context.Device.Id, twinTest.ModuleId, twinpatch.ToString(), currentTwin.ETag);
+                }
+
+                if (twinTest.TwinTest.ContainsKey("expectedReported") && twinTest.TwinTest["expectedReported"].Count > 0)
+                {
+                    using (var cts = new CancellationTokenSource(TimeSpan.FromSeconds(20)))
+                    {
+                        bool foundAtLeastOne = false;
+                        while (true)
+                        {
+                            await Task.Delay(TimeSpan.FromSeconds(10), cts.Token);
+
+                            currentTwin = await this.context.RegistryManager.GetTwinAsync(this.context.Device.Id, twinTest.ModuleId, cts.Token);
+
+                            foreach (KeyValuePair<string, string> expectedReportedProperty in twinTest.TwinTest["expectedReported"])
+                            {
+                                if (currentTwin.Properties.Reported.Contains(expectedReportedProperty.Key))
```

Comment:
this will only verify part of expected reported properties are matched, but may not be all of them.  We should fail the test if expected report property is not in current twin, right?

Reply:
This for loops verify all expectedReportedProperty. 
We are assuming that, if you have 1, you will have all, so we just need to check it's content. If we have any content failing we will fail the test (Raise an exception).

Reply:
With the new changes this is not applicable anymore.